library/ui/src/main/java/com/google/android/exoplayer2/ui/PlayerView.java

```diff
@@ -1246,15 +1248,29 @@ private void updateForCurrentTrackSelections(boolean isNewPlayer) {
   }
 
   private boolean setArtworkFromMetadata(Metadata metadata) {
+    boolean isArtworkSet = false;
+    int currentPicType = -1;
     for (int i = 0; i < metadata.length(); i++) {
       Metadata.Entry metadataEntry = metadata.get(i);
+      int picType;
+      byte[] bitmapData;
       if (metadataEntry instanceof ApicFrame) {
-        byte[] bitmapData = ((ApicFrame) metadataEntry).pictureData;
+        bitmapData = ((ApicFrame) metadataEntry).pictureData;
+        picType = ((ApicFrame) metadataEntry).pictureType;
+      } else if (metadataEntry instanceof PictureFrame) {
+        bitmapData = ((PictureFrame) metadataEntry).pictureData;
+        picType = ((PictureFrame) metadataEntry).pictureType;
+      } else {
+        continue;
+      }
```

Comment:
Given multiple pictures where none are FRONT_COVER, I think it's probably preferable to prefer the first one, where-as this code will prefer the last. Maybe logic should be more like (disclaimer: untested):

```
// Prefer the first front cover picture, or else the first picture of any type.
if (currentPictureType == -1 || pictureType == PICTURE_TYPE_FRONT_COVER) {
  Bitmap bitmap = BitmapFactory.decodeByteArray(bitmapData, 0, bitmapData.length);
  isArtworkSet = setDrawableArtwork(new BitmapDrawable(getResources(), bitmap));
  currentPictureType = pictureType;
  if (currentPictureType == PICTURE_TYPE_FRONT_COVER) {
    break; // We've found the front cover.
  }
}
```

Reply:
Done